config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleDocumentBuilder.java

```diff
@@ -35,6 +39,22 @@ public Element build(Document document, List<Entity> entities) {
         return bundle;
     }
 
+    public Element buildDeleteBundle(Document document, List<Entity> entities) {
+        removeEntitiesForDeleteBundle(entities); // Filter entities for delete bundle based on type
+
+        final Element references = document.createElement(REFERENCES);
+        final Element mappings = document.createElement(MAPPINGS);
+
+        final Element bundle = createElementWithChildren(document, BUNDLE, references, mappings);
+
+        entities.forEach(e -> {
+            e.setMappingAction(MappingActions.DELETE); // Set Mapping Action to DELETE
```

Comment:
Document builder should not change entities, it should read entities to build document
handle this in BundleEntityBuilder itself

Reply:
Fixed.